public/app/features/trails/MetricSelect/MetricSearchTermsVariable.ts

```diff
@@ -0,0 +1,89 @@
+import { VariableHide } from '@grafana/data';
+import {
+  SceneVariable,
+  SceneObjectBase,
+  VariableValue,
+  SceneVariableState,
+  SceneVariableValueChangedEvent,
+  SceneObjectUrlSyncConfig,
+  SceneObjectUrlValues,
+  SceneObjectWithUrlSync,
+  sceneGraph,
+  AdHocFiltersVariable,
+} from '@grafana/scenes';
+
+import { VAR_FILTERS, VAR_METRIC_SEARCH_TERMS } from '../shared';
+
+import { createPromRegExp } from './util';
+
+export interface MetricsSearchTermsVariableState extends SceneVariableState {
+  terms: string[];
+}
+
+export class MetricSearchTermsVariable
+  extends SceneObjectBase<MetricsSearchTermsVariableState>
```

Comment:
I fail to see why this needs to be a new type of variable, why not just use a CustomVariable?

Reply:
I'll see if I can get everything I really need via `CustomVariable`.

Reply:
What I need is a variable that stores a list of search terms that has been typed in, but interpolates these differently:
- When persisting the search terms in the URL, I just want to interpolate this as a simple comma-separated list (for URL readability -- not a hard requirement, any simple format would do)
- When requesting a list of metric names that match these search terms, I want to this variable to be interpolated into a regular expression that the request to Prometheus will make use of

I was struggling to find a way to override the formatting as I do in `getValue()`. It seems I can provide `value` with a formatter, but it seems I would have to provide a formatter function each time I set state `value`. 

Looking more closely at CustomVariable, it just looks like a list of preset custom options you could select one of more of. If that's the case, it might not be what I'm looking for.

Reply:
Maybe I can extend the TextBoxVariable (sic?) instead. Maybe I can even render it on the MetricSelectScene (sic?) Directly.

Reply:
@torkelo 
I have simplified much of the code here by extending `TextBoxVariable` in 2ecb62a.